### DISCUSSION TITLE
StructuralCurveMoment: "Value 1" and "Value 2" mandatory fields?

### DISCUSSION THREAD

**Discussion by scia-nv** [StructuralAnalysisFormat/StructuralAnalysisFormat-Doc#31] [Q&A] [answered]
There is inconsistency found in SAF spec.

For StructuralCurveMoment we have two options for distribution "Uniform" or "Trapez".
In SAF specification, we are defining for both distribution cases "Value 1" and "Value 2" as required.

There is an inconsistency in comparison to similar object StructuralCurvaAction, where "Value 2" is required only for "Distribution" = Trapez.

See samples of objects according to SAF specification for version SAF 2.0.0 below. With the release of SAF version 2.0.0, the "Value 2" became mandatory only for "Distribution" = Trapez for object StructuralCurveAction, but remain mandatory for both distributions types for StructuralCurveMoment.

StructuralCurveMoment according to spec:
| **Name** | **Type**       | Force action | Distribution | Direction | Value 1 \[kNm/m\] | Value 2 \[kNm/m\] | Other columns .. |
| ---- | ---------- | ------------ | ------------ | --------- | ----------------- | ----------------- | ---------------- |
| LF1  | Hoar frost | On beam      | Uniform      | Mx        | 1.00              | 1.00               | …                |
| LF2  | Hoar frost | On rib       | Trapez       | Mx        | 1.00              | 2.00              | …                |

StructuralCurveAction according to spec:
| Name | Type     | Force action | Distribution | Direction | Value 1 \[kN/m\] | Value 2 \[kN/m\] | Other columns |
| ---- | -------- | ------------ | ------------ | --------- | ---------------- | ---------------- | ------------- |
| LF1  | Standard | On beam      | Uniform      | Z         | \-1.00           |                  | …             |
| LF2  | Standard | On beam      | Trapez       | Z         | \-1.00           | \-2.00           | …             |

Please, let us know how the objects were implemented in your software. 

**Proposed solution**
1) Check the implementation status of interested parties.
2) The proposed solution is to make "Value 2" for StructuralCurveMoment required only for "Distribution" = Trapez in new major version of SAF.

**Comment by dschuermans** [comment, marked as answer]
Changing the required condition doesn't require a major release.
It doesn't break anything
This change will be made in the SAF 2.1.0 specification, as it really is a trivial change for anyone to implement (if they are even performing validation)
The users of the SDK will already have it implemented like this
